### PR TITLE
Blog app: add API endpoints

### PR DESCRIPTION
**Pull request by David-Kasilia**
To Highlight 💡 
- Create an API endpoint to list all posts for a user.
- `localhost:3000/api/v1/users/`
- Create an API endpoint to list all comments for a user's post.
-  `localhost:3000/api/v1/users/1/posts/3/comments`
- Create an API endpoint to add a comment to a post. Remember that the comment owner is the user that makes it; it should **not** be set by passing a parameter.
- Add authentication to your API endpoints.
- Your API endpoints should receive JSON and respond to JSON as well.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -11,8 +11,9 @@
   
   namespace :api do
     namespace :v1 do
-      post :auth, to: "authentication#create"
+      post :auth, to: "authentication#login"
       resources :users do
+        post :create, to: "users#create"
         resources :posts do
           resources :comments
```
Comment (chrystalme) [inline config/routes.rb]: - Kindly include the following missing routes - /login and /register or /signup

File: app/controllers/api/v1/comments_controller.rb [hunk 2]
```diff
@@ -1,27 +1,33 @@
...
     end
   end
-  
\ No newline at end of file
+
+  private
+
+  def comment_params
+    params.permit(:text)
+  end
```
Comment (chrystalme) [inline app/controllers/api/v1/comments_controller.rb:32]: While trying to use the ```Post - /users/:user_id/posts/:id/comments```, I could not post a comment. Kindly check that you are permitting the appropriate data to complete the post request.

**Review by chrystalme — CHANGES_REQUESTED** [review]
Great job @David-Kasilia and @Paul-js-hub, 

You have done a good job so far!
However, there are some issues that you still need to work on to prepare your project for further evaluation
## Highlights 💡 
✔️ Linter configured and working properly.
✔️ Gitflow is used correctly.
✔️ Project well documented.
✔️ ```GET -- /api/v1/users/:id/posts/:id/comment and   /api/v1/users/:id/posts ``` works fine
## Suggested changes ♻️ 
 - Kindly check that ```/login and /register or /signup``` routes are present in your config/routes.rb file
 - Also kindly check your comments controllers - comment_params. I get this error ```{
    "status": "ERROR",
    "message": "Comment Not Created"
} ```
_Check the comments under the review._


## Optional Changes
 - None

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Also, you can reach me on [Twitter](https://twitter.com/afam_ifeanyi) or [Slack](https://microverse-community.slack.com/team/U019ZC99KFA). Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/api/v1/comments_controller.rb (hunk 2 above)
Comment (David-Kasilia) [reply app/controllers/api/v1/comments_controller.rb:32]: Hello @chrystalme. Thank you for your review. I would like to show you that the post comment is working well. Below is a screenshot of the result. I tested it after seeing this review.
adddsadsadadsadsadasdadsa

File: config/routes.rb (hunk 1 above)
Comment (David-Kasilia) [reply config/routes.rb]: Added The routes /login and /signup
`POST   /api/v1/signup`
`POST   /api/v1/login`

fsdfsdfsdf

**Review by tresorsawasawa — APPROVED** [review]
# STATUS: APPROVE :heavy_check_mark: :heavy_check_mark:  :champagne: 

Hi @David-Kasilia and @Paul-js-hub 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 :clap: :clap: :100: 

nodding-yes

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Your README file is professional :heavy_check_mark: 

## Optional suggestions
_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**